optimum_benchmark/backends/pytorch.py

```diff
@@ -86,179 +165,172 @@ def configure(self, config: PyTorchConfig) -> None:
+                device_map=self.config.device_map
+                if self.config.device_map is not None
+                else self.device,
             )
 
-            LOGGER.info(f"\t+ Moving model to device: {self.device}")
-            self.pretrained_model.to(self.device)
-            self.pretrained_model.tie_weights()
-
-        else:
-            LOGGER.info(f"\t+ Materializing model on device: {self.device}")
-            self.pretrained_model.to_empty(device=self.device)
-
-            LOGGER.info("\t+ Randomizing model weights")
-            randomize_weights(self.pretrained_model)
-            self.pretrained_model.tie_weights()
-
-    def load_model_from_pretrained(self, config: PyTorchConfig) -> None:
-        LOGGER.info(
-            f"\t+ Loading pretrained model weights in dtype: {config.torch_dtype} on device: {self.device}"
-        )
-        if self.task not in ["stable-diffusion", "stable-diffusion-xl"]:
-            kwargs = {}
-            if config.load_in_8bit:
-                kwargs["load_in_8bit"] = config.load_in_8bit
-                kwargs["llm_int8_threshold"] = 0
-            elif config.load_in_4bit:
-                kwargs["load_in_4bit"] = config.load_in_4bit
-            
-            if config.device_map:
-                kwargs["device_map"] = config.device_map if config.device_map is not None else self.device
-
-                self.pretrained_model = self.automodel_class.from_pretrained(
-                    pretrained_model_name_or_path=self.model,
-                    torch_dtype=self.torch_dtype,
-                    **kwargs,
-                    **self.hub_kwargs,
-                )
-            else:
-                # When a device_map is not specified, we do not rely on accelerate to load the load and rather try PyTorch-native context.
-                with self.device:
-                    self.pretrained_model = self.automodel_class.from_pretrained(
-                        pretrained_model_name_or_path=self.model,
-                        torch_dtype=self.torch_dtype,
-                        **kwargs,
-                        **self.hub_kwargs,
-                    )
-        else:
-            self.pretrained_model = self.automodel_class.from_pretrained(
-                pretrained_model_name_or_path=self.model,
-                torch_dtype=self.torch_dtype,
-                device_map=config.device_map,
-                **self.hub_kwargs,
-            )
-            if config.device_map is None:
-                # Diffusers does not support device_map being a torch.device, thus if not provided, move to device here.
-                self.pretrained_model.to(self.device)
-
-    def prepare_for_profiling(
-        self,
-        input_names: List[str],
-        input_shapes: Dict[str, int],
-    ) -> None:
+    def prepare_for_profiling(self, input_names: List[str]) -> None:
         LOGGER.info("Preparing model for profiling")
```

Comment:
It could make sense to support pytorch profiler as well in the future (https://pytorch.org/tutorials/recipes/recipes/profiler_recipe.html) that does not rely on torch.fx so is easier to use (but serves a different purpose)